recipes/tesseract/all/conanfile.py

```diff
@@ -99,6 +106,14 @@ def validate(self):
                 f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
             )
 
+        leptonica = self.dependencies["leptonica"]
+        leptonica_with_tiff = leptonica.options.with_tiff
+        self_with_tiff = self.options.get_safe("with_tiff", leptonica_with_tiff)
+        if self_with_tiff != leptonica_with_tiff:
```

Comment:
This PR was not approved when you first submitted it because @czoido and I found this a bit weird, but we still need to have a discussion around this pattern

Reply:
A bit more context that I should have written before marking this ready for review:
There was a conan v2 only error when just trying to bump versions (c.f. the first bot report).
My understanding was that when calling `find_package(TIFF)`, cmake would find it (even though it's not declared as transitive lib or header in leptonica).

As mentioned in a comment in the recipe, libtiff is only used in an executable (that's also being packaged here). It's only used to configure error/warning handling on windows for actual calls to libtiff which will happen in leptonica (so only relevant when building leptonica with_tiff=True).

So, ideally, there should not be an option `with_tiff` but just inherit this from leptonica on Windows (along with the version if possible). I think I saw the same kind of thing being asked in another PR but forgot which one.

Given that's not possible as far as I understand, the best next thing IMHO was to add a `with_tiff` option for windows builds and making sure that it was the same as the leptonica one in validate.

Reply:
I have been checking the upstream CMakeLists and I think that conditionally adding libtiff, that is what tesseract does for windows versions higher than 5.0.0 should be good enough. Let me push some changes to this PR, and I think we can get rid of the extra complexity of managing those options.